### DISCUSSION TITLE
Redirect user to different content other than "OIDCRedirectURI"

### DISCUSSION THREAD

**Discussion by mzico** [OpenIDC/mod_auth_openidc#875] [Q&A]
Hi, 

To present my scenario: 

 - `https://myapp.com/login` 
 - `https://myapp.com/content` 
 
 My OIDCRedirectURI is pointing `https://myapp.com/login` for authentication. ( Meaning: `https://myapp.com/login` --> OIDC Provider for authentication --> successful authentication ). 
 
 Now, I would like my user to land into `https://myapp.com/content` instead of `https://myapp.com/login` after successful authentication. Is that possible? 
 
 According to mod_auth_oidc documentation: 
 
 ```
 # (Mandatory)
# The redirect_uri for this OpenID Connect client; this is a vanity URL
# that must ONLY point to a path on your server protected by this module
# but it must NOT point to any actual content that needs to be served.
# You can use a relative URL like /protected/redirect_uri if you want to
# support multiple vhosts that belong to the same security domain in a dynamic way
#OIDCRedirectURI https://www.example.com/protected/redirect_uri
```

**Comment by zandbelt** [comment]
that's the default behavior, just protect `/content` with `AuthType openid-connect` and it should work like you describe when you access `/content`

**Reply by ryanwilliamnicholls** [reply]
What if you want all logins to go to `/content` after authentication regardless if they accessed `/content` or not?

**Reply by zandbelt** [reply]
that's more of an Apache question than an authentication question: you could create a [permanent redirect](https://httpd.apache.org/docs/current/mod/mod_alias.html#redirectmatch) or configure a [condition](https://fedingo.com/how-to-redirect-in-apache-based-on-cookie/) to check for the session cookie and, if it does not exist, redirect to /content

**Reply by Clivern** [reply]
@zandbelt I came across this and it works perfectly fine so thanks. but it seems to ignore the anchor links something like 
http://example.com/sub1/sub2/sub3/#sub4/d?x=y&z=k

will send back to

http://example.com/sub1/sub2/sub3

Ignoring whatever after the #..

I think it has to do with this function https://github.com/OpenIDC/mod_auth_openidc/blob/master/src/util.c#L767-L793 not capturing anchor links

**Reply by zandbelt** [reply]
no, fragments never reach the server again: https://stackoverflow.com/questions/940905/can-i-read-the-hash-portion-of-the-url-on-my-server-side-application-php-ruby

**Reply by Clivern** [reply]
Hmmm interesting! Thanks @zandbelt

**Reply by MuhammedGenco94** [reply]
Thanks, this conversation was helpful for my problem, too.
The **fragments** after Hash `#` were the root cause
